src/components/learner-credit-management/assignments-status-chips/WaitingForLearner.jsx

```diff
@@ -6,26 +6,45 @@ import { getConfig } from '@edx/frontend-platform/config';
 
 import BaseModalPopup from './BaseModalPopup';
 import { ASSIGNMENT_ENROLLMENT_DEADLINE } from '../data';
+import EVENT_NAMES from '../../../eventTracking';
 
-const WaitingForLearner = ({ learnerEmail }) => {
+const WaitingForLearner = ({ learnerEmail, trackEvent }) => {
   const [isOpen, open, close] = useToggle(false);
   const [target, setTarget] = useState(null);
+  const {
+    BUDGET_DETAILS_ASSIGNED_DATATABLE_CHIP_WAITING_FOR_LEARNER,
+    BUDGET_DETAILS_ASSIGNED_DATATABLE_CHIP_WAITING_FOR_LEARNER_HELP_CENTER,
+  } = EVENT_NAMES.LEARNER_CREDIT_MANAGEMENT;
+
+  const openChipModal = () => {
```

Comment:
nit/suggestion: wondering if its worth DRYing up some of the repeated logic around these functions. Would it make sense to create a custom hook that could be re-used across all chips?

```jsx
const {
  openChipModal,
  closeChipModal,
  isChipModalOpen,
} = useAssignmentStatusChip({
  trackEvent,
  chipInteractionEventName: BUDGET_DETAILS_ASSIGNED_DATATABLE_CHIP_WAITING_FOR_LEARNER,
  chipHelpCenterEventName: BUDGET_DETAILS_ASSIGNED_DATATABLE_CHIP_WAITING_FOR_LEARNER_HELP_CENTER,
});

const useAssignmentStatusChip = ({ chipInteractionEventName, chipHelpCenterEventName, trackEvent }) => {
  const [isChipModalOpen, open, close] = useToggle(false);
  const openChipModal = () => {
    open();
    // Note: could hardcode true given this function *always* opens
    trackEvent(chipInteractionEventName, { isOpen: true });
  };
  const closeChipModal = () => {
    close();
        // Note: could hardcode false given this function *always* closes
    trackEvent(chipInteractionEventName, { isOpen: false });
  };

  const helpCenterTrackEvent = () => {
    trackEvent(chipHelpCenterEventName);
  };

  return {
    isChipModalOpen,
    openChipModal,
    closeChipModal,
  };
};
```

Reply:
The abstractions worked out great 👍🏽  way cleaner implementation for toggle-able events